build.gradle

```diff
@@ -224,6 +224,33 @@ subprojects {
             }
         }
     }
+
+    File dexAllDir = new File(project.buildDir, "dex-all")
+    task copyDependencies(type: Copy) {
+        from configurations.runtime
+        into dexAllDir
+    }
+
+    task copyLib(type: Copy) {
+        from jar.destinationDir
+        into dexAllDir
+    }
+
+    task dexAll(type: Exec, dependsOn: [copyDependencies, copyLib]) {
+        workingDir dexAllDir
+        commandLine 'bash', '-e', '-c', """
+            ls -1 . > dexInputList.txt
+            dx.exec --dex --statistics --num-threads=4 --core-library --output=./file.dex --input-list=dexInputList.txt >dex.log 2>&1
+            grep 'method id:.*' dex.log
+        """.replace("dx.exec",(System.getProperty("dx.exec") == null) ? "dx" : System.getProperty("dx.exec"))
+    }
+
+    task methodCount(dependsOn: [build, copyDependencies, copyLib, dexAll]) {
```

Comment:
@tomblench fixed the dependencies and made these other improvements in 42dbe4b:
* Created `AndroidExec` task type to add android tools to the path reading
`sdk.home` or `ANDROID_HOME`.
* Removed `dx.exec` property and used `AndroidExec` type instead.
* Fixed dependencies including using `assemble` instead of `build` to avoid
running tests if just doing `methodcount`.
* Added input/output directories to `dexAll` task to avoid running twice
unnecessarily.

Unfortunately sharing the `AndroidExec` type with the AndroidTest/app project is proving problematic because they don't share a root build. I think I'd like to merge what I have here and consider the sharing later (possibly with the log collection as that is going to need android commands as well).